### PR TITLE
feat: add `flox help` command

### PR DESCRIPTION
**Pull request by ysndr**
closes https://github.com/flox/flox/issues/425

### CODE REVIEW

**Review by ghudgins — CHANGES_REQUESTED** [review]
image

don't list the help command in help. there's an additional `Additional Commands` section. if you need to have it, put it in the existing `Additional Commands` section

**Comment by ysndr** [comment]
@ghudgins, done hidden now

File: cli/flox/src/commands/mod.rs [hunk 1]
```diff
@@ -243,12 +244,35 @@ impl FloxArgs {
...
+impl Help {
+    fn handle(self) {
+        let mut args = Vec::from_iter(self.cmd.as_deref());
+        args.push("--help");
+
+        match flox_cli().run_inner(&*args) {
+            Ok(_) => unreachable!(),
+            Err(ParseFailure::Completion(comp)) => print!("{comp}"),
+            Err(ParseFailure::Stdout(doc, _)) => info!("{doc}"),
+            Err(ParseFailure::Stderr(err)) => error!("{err}"),
```
Comment (mkenigs) [inline cli/flox/src/commands/mod.rs:271]: nonblocking: I don't really understand what the Completion or Stderr arms are doing - are those reachable?

**Review by mkenigs — APPROVED** [review]

File: cli/flox/src/commands/mod.rs (hunk 1 above)
Comment (ysndr) [reply cli/flox/src/commands/mod.rs:271]: I confirmed stderr might be at some point but isn't now
The completions might bereached if you put the right flags behind -- but should be catched earlier

Generally neither should be reachable in normal circumstances

**Pull request merged**